cmd/cert-checker/main.go

```diff
@@ -490,7 +504,12 @@ func (c *certChecker) checkCert(ctx context.Context, cert *corepb.Certificate) (
 			}
 		}
 	}
-	return dnsNames, problems
+
+	sans := slices.Clone(parsedCert.DNSNames)
+	for _, ip := range parsedCert.IPAddresses {
+		sans = append(sans, ip.String())
+	}
+	return sans, problems
```

Comment:
You're outside of the `else` clause that starts on `349` and ends on `508`. `parsedCert` may be nil here.

Interestingly enough we also have a similar issue on `468` where we do `p, err := x509.ParseCertificate(cert.Der)` but make no effort to use an `else` clause at all before calling `c.kp.GoodKey(ctx, p.PublicKey)`

Reply:
Good catch. I've rearranged things to avoid needing this huge whole-function-long `else` stanza.